src/node/structural_hash.cc

```diff
@@ -360,6 +360,46 @@ struct ADTObjTrait {
 
 TVM_REGISTER_REFLECTION_VTABLE(runtime::ADTObj, ADTObjTrait);
 
+struct ModuleNodeTrait {
+  static constexpr const std::nullptr_t VisitAttrs = nullptr;
+  static constexpr std::nullptr_t SHashReduce = nullptr;
+  static constexpr std::nullptr_t SEqualReduce = nullptr;
+};
+
+TVM_REGISTER_REFLECTION_VTABLE(runtime::ModuleNode, ModuleNodeTrait)
+    .set_creator([](const std::string& blob) {
+      dmlc::MemoryStringStream mstrm(const_cast<std::string*>(&blob));
+      support::Base64InStream b64strm(&mstrm);
+      b64strm.InitPosition();
+      // retrieve derived type key 
+      dmlc::Stream* stream = static_cast<dmlc::Stream*>(&b64strm);
+      std::string derived_type;
+      stream->Read(&derived_type);
+      // pick up the deserializer
+      std::string load_func_key = "runtime.module.loadbinary_"+derived_type;
+      const auto* load_func = runtime::Registry::Get(load_func_key);
+      ICHECK(load_func) << load_func_key << " is not registered.";
+      // deserialize
+      runtime::Module rtmod = (*load_func)(&b64strm);
+      return RefToObjectPtr::Get(rtmod);
+    })
+    .set_repr_bytes([](const Object* n) -> std::string {
+      const auto* rtmod = static_cast<const runtime::ModuleNode*>(n);
+      std::string blob;
+      dmlc::MemoryStringStream mstrm(&blob);
+      support::Base64OutStream b64strm(&mstrm);
```

Comment:
This is not sufficient, we will need to collect the import tree, as in normal export, so not only module itself get exported, but also the imported modules

Reply:
Thanks for the catch! This is addressed by using https://github.com/apache/tvm/blob/main/src/runtime/dso_library.cc#L152 which also collects import tree.